test/user.spec.js

```diff
@@ -67,13 +67,87 @@ describe('POST /signup - create users', () => {
       lastName: '',
       password: 'abc12345'
     };
+    createUser(userDataUncompletedFields).then(res => {
+      expect(res.status).toBe(400);
+      expect(res.body.internal_code).toBe('bad_request_error');
+      done();
+    });
+  });
+});
+
+describe('POST /users/sessions  - signIn user', () => {
+  const user = {
+    name: 'hector',
+    lastName: 'gonzalez',
+    email: 'hector@wolox.com.ar',
+    password: 'abc12345'
+  };
+  it('should succeed returning the generated token', done => {
+    const signInData = {
+      email: 'hector@wolox.com.ar',
+      password: 'abc12345'
+    };
+    createUser(user).then(res => {
+      expect(res.status).toBe(201);
+      request(app)
+        .post('/users/sessions')
+        .send(signInData)
+        .then(response => {
+          expect(response.status).toBe(200);
+          expect(response.body.token).toBeDefined();
+          done();
+        });
+    });
+  });
+
+  it('should fail returning 400 code error because uncompleted fields', done => {
+    const signInData = {
+      email: 'hectorwolox.com.ar',
+      password: ''
+    };
     request(app)
-      .post('/users')
-      .send(userDataUncompletedFields)
+      .post('/users/sessions')
+      .send(signInData)
       .then(res => {
         expect(res.status).toBe(400);
         expect(res.body.internal_code).toBe('bad_request_error');
         done();
       });
   });
+
+  it('should fail returning 400 code error because invalid password', done => {
+    const signInData = {
+      email: 'hector@wolox.com.ar',
+      password: 'asdasdasd5'
+    };
+    createUser(user).then(res => {
+      expect(res.status).toBe(201);
+      request(app)
+        .post('/users/sessions')
+        .send(signInData)
+        .then(response => {
+          expect(response.status).toBe(400);
+          expect(response.body.internal_code).toBe('bad_request_error');
+          done();
+        });
+    });
+  });
+
+  it('should fail returning 404 code error because the user with the requested email was not found', done => {
+    const signInData = {
+      email: 'hector@wolox.com.ar',
+      password: 'asasdasdasds3'
+    };
+    createUser(user).then(res => {
+      expect(res.status).toBe(201);
```

Comment:
Don't test create endpoint again, you can create user directly in database (or create a utils/helper for do this), because if this endpoint changes some functionality, all sign-in test would fail too

Reply:
If you want you could use something like [factory-girl](https://github.com/aexmachina/factory-girl)